src/cert_provider/cert_provider_test.cc

```diff
@@ -0,0 +1,503 @@
+ * Just  a fleet CA without a fleet private key.
+ * Just a fleet private key without a fleet CA
+ * Neither `--target` nor `--local` is specified
+ *
+ * Checks actions:
+ *
+ *  - [x] Abort if fleet CA is provided without fleet private key
+ *  - [x] Abort if fleet private key is provided without fleet CA
+ */
+
+TEST_F(AktualizrCertProviderTest, IncompleteFleetCredentials) {
+  const std::string expected_error_msg = "fleet-ca and fleet-ca-key options should be used together\n";
+
+  {
+    DeviceCredGenerator::ArgSet args;
+
+    args.fleetCA = test_args_.fleet_ca_cert;
+    // args.fleetCAKey = test_args_.fleet_ca_private_key;
+    args.localDir = test_args_.test_dir;
+
+    device_cred_gen_.run(args);
+
+    ASSERT_EQ(device_cred_gen_.lastExitCode(), 1);
+    ASSERT_EQ(device_cred_gen_.lastStdErr(), expected_error_msg) << device_cred_gen_.lastStdErr();
+  }
+
+  {
+    DeviceCredGenerator::ArgSet args;
+
+    // args.fleetCA = test_args_.fleet_ca_cert;
+    args.fleetCAKey = test_args_.fleet_ca_private_key;
+    args.localDir = test_args_.test_dir;
+
+    device_cred_gen_.run(args);
+
+    ASSERT_EQ(device_cred_gen_.lastExitCode(), 1);
+    ASSERT_EQ(device_cred_gen_.lastStdErr(), expected_error_msg) << device_cred_gen_.lastStdErr();
+  }
+
+  {
+    DeviceCredGenerator::ArgSet args;
+
+    args.fleetCA = test_args_.fleet_ca_cert;
+    args.fleetCAKey = test_args_.fleet_ca_private_key;
+    // args.localDir = test_args_.test_dir;
+
+    device_cred_gen_.run(args);
+
+    ASSERT_EQ(device_cred_gen_.lastExitCode(), 1);
+    ASSERT_NE(device_cred_gen_.lastStdErr().find("Both a local dir and a target are not specified,"
+                                                 " thus the resultant cert and key will be lost"),
+              std::string::npos)
+        << device_cred_gen_.lastStdErr();
+  }
+}
+
+/**
+ * Verifies usage of the paths from a config file which is specified via `--config` param.
+ * The resultant files's path, private key and certificate files, must correspond to what is specified in the config.
+ *
+ * Verifies cert_provider's output if both `--directory` and `--config` params are specified
+ *
+ * Checks actions:
+ *
+ *  - [x] Use file paths from config if provided
+ */
+TEST_F(AktualizrCertProviderTest, ConfigFilePathUsage) {
+  const std::string base_path = "my_device_cred";
+  const std::string private_key_file = "my_device_private_key.pem";
+  const std::string cert_file = "my_device_cert.pem";
+
+  Config config;
+  config.import.base_path = base_path;
+  config.import.tls_pkey_path = BasedPath(private_key_file);
+  config.import.tls_clientcert_path = BasedPath(cert_file);
+
+  auto test_conf_file = tmp_dir_ / "conf.toml";
+  boost::filesystem::ofstream conf_file(test_conf_file);
+  config.writeToStream(conf_file);
+  conf_file.close();
+
+  // for some reason the cert-provider uses the config's base path as suffix (the same as the 'directory' param) of the
+  // local dir
```

Comment:
This is for when we use it inside bitbake. According to the config, the import dir will be `/var/sota/import`, but when we prepare the image, it will have a really long prefix.